File: resources/todo-go/validate.go
import "strings"

// Run validates todo data before saving
func Run(ctx *EventContext) error {
	// Validate title
	title, ok := ctx.Data["title"].(string)
	if !ok || strings.TrimSpace(title) == "" {
		ctx.Cancel("Title is required", 400)
		return nil
	}

	if len(title) > 200 {
		ctx.Cancel("Title is too long (max 200 characters)", 400)
		return nil
	}

	// Validate priority if provided
	if priority, exists := ctx.Data["priority"]; exists {
		if priorityNum, ok := priority.(float64); ok {
			if priorityNum < 1 || priorityNum > 5 {
				ctx.Cancel("Priority must be between 1 and 5", 400)
				return nil
			}
		}
	}

	// Trim whitespace from title
	ctx.Data["title"] = strings.TrimSpace(title)

	// Log validation success using proper logging
	ctx.Log("Todo validation completed", map[string]interface{}{
		"title":  title,
		"action": "validate",
	})
	return nil
}

File: resources/todo-js/validate.js
// Run function for todo validation using new context pattern
function Run(context) {
    context.log("Todo validation executing with Run(context) pattern");
    
    // Validate title
    if (!context.data.title || context.data.title.length < 1) {
        context.cancel("Title is required", 400);
        return;
    }
    
    if (context.data.title.length > 200) {
        context.cancel("Title is too long (max 200 characters)", 400);
        return;
    }
    
    // Validate priority
    if (context.data.priority !== undefined && (context.data.priority < 1 || context.data.priority > 5)) {
        context.cancel("Priority must be between 1 and 5", 400);
        return;
    }
    
    context.log("Todo validation passed successfully");
}